audit/hashstructure.go

```diff
@@ -107,13 +107,13 @@ func Hash(salter *salt.Salt, raw interface{}) error {
 // the structure. Only _values_ are hashed: keys of objects are not.
 //
 // For the HashCallback, see the built-in HashCallbacks below.
-func HashStructure(s interface{}, cb HashCallback) (interface{}, error) {
+func HashStructure(s interface{}, cb HashCallback, ignoredKeys []string) (interface{}, error) {
```

Comment:
I think a good idea here would be to turn this into a struct which currently has one member. That way if we want to add support later to not-HMAC other values (that aren't in the request/response Data) we can do it without breaking the interface again. You can imagine for instance that rather than set hmac_accessor directly on backends at mount time, we turn it into something tunable per-audit-mount via sys/config, and then that gets passed in as a value in this struct.

Reply:
Right now `HashStructure()` is really only used on `s.Data` in the `Hash()` function. Hashing of the other values in auth, req, and resp objects is mainly done by `Hash()`, so we should consider passing of the struct in there. I can see hmac_accessor being moved to be per-audit-mount and passing that option into `Hash()` (currently logic is performed outside of it and cached/restored after hashing https://github.com/hashicorp/vault/blob/master/audit/format.go#L94).

Reply:
OK, leave as is.